website/www/site/layouts/shortcodes/highlight.html

```diff
@@ -12,20 +12,36 @@
 
 {{ $content := (trim .Inner "\n\r") | htmlUnescape | safeHTML }}
 {{ $ctx := . }}
-{{ with (.Get "class") }}
-    <div class={{ . }}>
-        {{ with ($ctx.Get "lang") }}
+{{ $py := .Get "py" }}
+{{ with (.Get "language") }}
+    <div class='{{ printf "language-%s" . }} snippet'>
+        <div class="notebook-skip code-snippet">
+            {{ with $py }}
+                <a target="_blank" type="button" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Run code" href="https://colab.research.google.com/github/{{ $ctx.Site.Params.branch_repo }}/{{ $ctx.Get `notebook` }}-py.ipynb">
+                    <img src="/images/run-icon.svg"/>
+                </a>
+                <a target="_blank" type="button" data-bs-toggle="tooltip" data-bs-placement="bottom" title="View source code" href="https://github.com/{{ $ctx.Site.Params.branch_repo }}/{{ $ctx.Get `py` }}">
+                    <img src="/images/code-icon.svg"/>
+                </a>
+                <a class="copy" type="button" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Copy to clipboard">
+                    <img src="/images/copy-icon.svg"/>
+                </a>
+            {{ else }}
+                <a class="copy" type="button" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Copy to clipboard">
+                    <img src="/images/copy-icon.svg"/>
+                </a>
+            {{ end }}
             {{ highlight $content . "" }}
-        {{ else }}
-            {{ highlight $content "" "" }}
-        {{ end }}
+        </div>
     </div>
 {{ else }}
-    {{ with (.Get 0) }}
-        <div class={{ printf "language-%s" . }}>
-            {{ highlight $content . "" }}
+    <div class="snippet">
+        <div class="notebook-skip code-snippet without_switcher">
+            <a class="copy" type="button" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Copy to clipboard">
+                <img src="/images/copy-icon.svg"/>
+            </a>
+            {{ highlight $content "" "" }}
         </div>
-    {{ else }}
-        {{ highlight $content "" "" }}
-    {{ end }}
+    </div>
```

Comment:
There are a few odd things about the way this code snippet template renders for me:
![image](https://user-images.githubusercontent.com/675055/106643897-34e99380-653f-11eb-8bf1-dc2bb77b6fe1.png)

- There's an extra horizaontal line at the top of the snippet that extends off to the right
- It looks like there's a 1px gap between the tabs and the top of the snippet, which looks weird to me
- There's a lot of empty space before the first line of code (and to a lesser extent, after the last line of code), I think this looks pretty clunky, especially for snippets that have just one line.

Reply:
I'd say the only one of these that's a blocker is the horizontal line extending off to the right.

Reply:
Ok that makes sense.

Reply:
The last issue is resolved now, thanks